lang/cs/Org.Apache.REEF.Client/API/JobRequestBuilder.cs

```diff
@@ -18,32 +18,53 @@
 using System;
 using System.Collections.Generic;
 using Org.Apache.REEF.Common.Client.Parameters;
+using Org.Apache.REEF.Client.Common;
+using Org.Apache.REEF.Client.Common.Parameters;
 using Org.Apache.REEF.Tang.Annotations;
 using Org.Apache.REEF.Tang.Interface;
 using Org.Apache.REEF.Utilities.Logging;
+using System.IO;
+using System.Linq;
 
 namespace Org.Apache.REEF.Client.API
 {
     public sealed class JobRequestBuilder
     {
         private readonly AppParametersBuilder _appParametersBuilder = AppParametersBuilder.NewBuilder();
         private readonly JobParametersBuilder _jobParametersBuilder = JobParametersBuilder.NewBuilder();
+        private readonly IList<string> _globalAssemblies;
+        private readonly bool _defaultToGlobalAssembliesInWorkingDirectory;
+        private const string DLLFileNameExtension = ".dll";
+        private const string EXEFileNameExtension = ".exe";
+        private bool _globalAssembliesAdded = false;
 
         private JobRequestBuilder()
         {
         }
 
         [Inject]
-        internal JobRequestBuilder([Parameter(typeof(DriverConfigurationProviders))] ISet<IConfigurationProvider> configurationProviders)
+        internal JobRequestBuilder(
+            [Parameter(typeof(DriverConfigurationProviders))] ISet<IConfigurationProvider> configurationProviders,
+            [Parameter(typeof(GlobalAssemblies))] IList<string> globalAssemblies,
+            [Parameter(typeof(DefaultToGlobalAssembliesInWorkingDirectory))] bool defaultToGlobalAssembliesInWorkingDirectory)
         {
             AddDriverConfigurationProviders(configurationProviders);
+            _globalAssemblies = globalAssemblies;
+            _defaultToGlobalAssembliesInWorkingDirectory = defaultToGlobalAssembliesInWorkingDirectory;
         }
 
         /// <summary>
         /// Bake the information provided so far and return a IJobSubmission 
         /// </summary>
         public JobRequest Build()
         {
+            // If no global assemblies were added, optionally default to those in the working directory
+            if (!_globalAssembliesAdded && _defaultToGlobalAssembliesInWorkingDirectory && _globalAssemblies.Count == 0)
```

Comment:
This is now the logic for adding the current working directory:
- No assemblies have been added
- The parameter `DefaultToGlobalAssembliesInWorkingDirectory` has been set to true (`Tang` default)
- The injected global assemblies list is empty